pkg/retriever/retriever.go

```diff
@@ -282,6 +272,7 @@ func makeOnRetrievalEvent(
 		}
 
 		eventManager.DispatchEvent(event)
+		eventsCb(event)
```

Comment:
I'd be OK with this:

```suggestion
		if eventsCb != nil {
			eventsCb(event)
		}
```